core/chains/evm/log/broadcaster.go

```diff
@@ -417,11 +425,7 @@ func (b *broadcaster) eventLoop(chRawLogs <-chan types.Log, chErr <-chan error)
 			needsResubscribe = b.onChangeSubscriberStatus() || needsResubscribe
 
 		case blockNumber := <-b.replayChannel:
-			// NOTE: This ignores r.highestNumConfirmations, but it is
-			// generally assumed that this will only be performed rarely and
-			// manually by someone who knows what he is doing
-			b.backfillBlockNumber.SetValid(blockNumber)
-			b.logger.Debugw("Returning from the event loop to replay logs from specific block number", "blockNumber", blockNumber)
+			b.onReplayRequest(blockNumber)
```

Comment:
If we have it above, do we still need it here?

Reply:
Yes, because we only check above, but then sit and wait here. We still need to yield to replay also.